src/commands/mimic/mod.rs

```diff
@@ -0,0 +1,156 @@
+use crate::commands::mimic::{delete::*, set::*};
+use crate::pawthos::{
+    enums::embed_type::EmbedType,
+    structs::mimic::Mimic,
+    types::{Context, Embed, Reply, Result},
+};
+use crate::utils;
+use poise::serenity_prelude as serenity;
+use serenity::{AutocompleteChoice, ExecuteWebhook};
+mod delete;
+mod set;
+
+/// Returns AutocompleteChoices to the Mimic slash commands that request Mimic Autocompletes.
+async fn fetch_mimics(ctx: Context<'_>, partial: &str) -> Vec<AutocompleteChoice> {
+    ctx.data()
+        .with_user_read(ctx.author().id, |maybe_user| {
+            maybe_user
+                .map(|user| {
+                    user.mimics
+                        .iter()
+                        .filter(|m| m.name.starts_with(partial))
+                        .map(|m| AutocompleteChoice::new(m.name.clone(), m.name.clone()))
```

Comment:
Clippy would say use filter_map rather than filter + map, but this is probably more readable